configure.ac

```diff
@@ -192,34 +192,59 @@ AC_SUBST([QEMU_PATH], [$ac_cv_path_QEMU])
 AC_SUBST([CMDLINE], [$(cat $srcdir/data/kernel-cmdline)])
 
 #bundle path for functional-test
-DEFAULT_BUNDLE_TEST_DIR=$(cd "$srcdir/tests/functional/bundle" 2>/dev/null; pwd -P)
 AC_ARG_WITH([tests-bundle-path],
     [AS_HELP_STRING([--with-tests-bundle-path=[[BUNDLE-PATH]]],
             [path to oci bundle to execute functional tests])
     ],
     [
         BUNDLE_TEST_PATH=$(cd "$with_tests_bundle_path"; pwd -P)
         AC_SUBST([BUNDLE_TEST_PATH])
-    ],
-    [
-        AC_SUBST([BUNDLE_TEST_PATH], [$DEFAULT_BUNDLE_TEST_DIR])
     ]
 )
 
-DEFAULT_ROOTFS_PATH=$BUNDLE_TEST_PATH/rootfs
 AC_ARG_WITH([tests-rootfs-path],
     [AS_HELP_STRING([--with-tests-rootfs-path=[[ROOTFS-PATH]]],
         [path to rootfs for functional tests])
     ],
     [
         ROOTFS_PATH=$(cd "$with_tests_rootfs_path"; pwd -P)
         AC_SUBST([ROOTFS_PATH])
-    ],
-    [
-        AC_SUBST([ROOTFS_PATH], [$DEFAULT_ROOTFS_PATH])
     ]
 )
 
+AC_ARG_ENABLE(auto-bundle-creation,
+    AS_HELP_STRING([--disable-auto-bundle-creation], [do not
+        automatically create a bundle directory
+        if --with-tests-bundle-path= or
+        --with-tests-rootfs-path= are not
+        specified]),
+        [], [auto_bundle_creation=yes])
+
+AC_CHECK_PROG([MKTEMP_AVAILABLE], [mktemp], [yes], [no])
+AC_CHECK_PROG([DOCKER_AVAILABLE], [docker], [yes], [no])
+
+AS_IF([test x"$with_tests_rootfs_path" = x"" dnl
+       -a x"$with_tests_bundle_path" = x"" dnl
+       -a x"$disable_tests" = xno dnl
+       -a x"$disable_functional_tests" = xno dnl
+       -a x"$auto_bundle_creation" = xyes ], [
+       AS_IF([test x"$DOCKER_AVAILABLE" = x"yes" -a x"$MKTEMP_AVAILABLE" = x"yes"],
+           [
+               BUNDLE_TEST_PATH=$([sh data/make-bundle-dir.sh])
```

Comment:
That does look somewhat ugly. How about generating the bundle in a sub-directory of tests/?

An alternative could be generating the test bundle with its own Makefile rule: mid-term we could build binaries to include in the test bundle (eg. test signal handling with a binary printing the name of received signal) and track the dependencies to rebuild the bundle (which would include the Makefile itself).